py/server/deephaven/table.py

```diff
@@ -52,18 +52,35 @@
 _JPythonScriptSession = jpy.get_type("io.deephaven.integrations.python.PythonDeephavenSession")
 _j_script_session = jpy.cast(_JQueryScope.getScope(), _JUnsynchronizedScriptSessionQueryScope).scriptSession()
 _j_py_script_session = jpy.cast(_j_script_session, _JPythonScriptSession)
-_deephaven_pkg_prefix = __file__[:__file__.index("deephaven") + len("deephaven")]
+_deephaven_pkg_prefix = __file__
 
 
-def _set_query_scope():
-    for frame, filename, line_num, func, source_code, source_index in inspect.getouterframes(inspect.currentframe()):
-        if filename and filename.startswith(_deephaven_pkg_prefix):
-            continue
+@contextlib.contextmanager
+def _query_scope_ctx():
```

Comment:
Are we sure this should be internal-only?

Reply:
After reading the tests, it might be totally reasonable for the context to be private, since it looks like @jmao-denver has managed to get function level scope automagically, without users needing to do anything.

Reply:
it is written for the specific purpose of setting formula/filter evaluation scope, unless we expand these evaluation beyond table ops, it is appropriate to keep it internal for now